src/download.js

```diff
@@ -59,8 +62,8 @@ async function cachingFetchAndVerify (url) {
 
   const data = fs.readFileSync(cachedFilePath)
   const calculatedSha = await Hash.of(data)
-  if (calculatedSha !== filename) {
-    console.log(`Expected CID: ${filename}`)
+  if (calculatedSha !== cid) {
```

Comment:
Maybe make this more robust, so it does not matter what is the CID version:
```suggestion
  if (CID.equals(CID.parse(calculatedSha), cid)) {
```

Reply:
I think if we want to do that, we should just compare the multihash bytes?